ui/dashboardApp/layout/signin/index.js

```diff
@@ -172,19 +173,22 @@ function TiDBSignInForm({ registry }) {
 function App({ registry }) {
   return (
     <Root>
-      <div className={styles.container}>
-        <div className={styles.dialogContainer}>
-          <div className={styles.dialog}>
-            <TiDBSignInForm registry={registry} />
+      <Router>
+        <TopLoadingBar />
```

Comment:
How about making it another SingleSPA app that always exists on the page, or simply put it outside everything mounted by SingleSPA. In this way, it can listen SingleSPA global events gracefully without worrying about its own lifecycle.

Reply:
It may be a solution, let me have a try.

Reply:
Implement it by this way in PR #661 , much more graceful! will close this PR.